### PR TITLE
Reservations new create 

### PR DESCRIPTION
**Pull request by Belaxyx**
I've done the backend for the new and index methods of the controller Reservation 
However, I do not have create the view for the show in reservation
I still need to do the html and css for new.html.erb

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -10,9 +10,29 @@ def show
   end
 
   def new
+    @date = Time.now.strftime("%B %d, %Y")
+    @pedalo = Pedalo.find(params[:pedalo_id])
```
Comment (maxjost) [inline app/controllers/reservations_controller.rb]: I suggest adding a before_action method

File: app/controllers/reservations_controller.rb [hunk 2]
```diff
@@ -10,9 +10,29 @@ def show
...
+    @reservation = Reservation.new
+    # do some JS if time to display the total price
```
Comment (maxjost) [inline app/controllers/reservations_controller.rb]: this will have to be done in new.html.erb

File: app/controllers/reservations_controller.rb [hunk 3]
```diff
@@ -10,9 +10,29 @@ def show
...
   end
 
   def create
+    @reservation = Reservation.new(reservation_params)
+    @user = current_user
+    @pedalo = Pedalo.find(params[:pedalo_id])
```
Comment (maxjost) [inline app/controllers/reservations_controller.rb]: before_action ?

File: app/controllers/reservations_controller.rb [hunk 4]
```diff
@@ -10,9 +10,29 @@ def show
...
   def new
+    @date = Time.now.strftime("%B %d, %Y")
+    @pedalo = Pedalo.find(params[:pedalo_id])
+    @reservation = Reservation.new
+    # do some JS if time to display the total price
   end
 
   def create
+    @reservation = Reservation.new(reservation_params)
+    @user = current_user
```
Comment (maxjost) [inline app/controllers/reservations_controller.rb]: i would remove this line

File: app/controllers/reservations_controller.rb [hunk 5]
```diff
@@ -10,9 +10,29 @@ def show
...
+    @pedalo = Pedalo.find(params[:pedalo_id])
+    @reservation.pedalo_id = @pedalo.id
```
Comment (maxjost) [inline app/controllers/reservations_controller.rb]: improvement: @reservation.pedalo = @pedalo

File: app/controllers/reservations_controller.rb [hunk 6]
```diff
@@ -10,9 +10,29 @@ def show
...
+    @reservation.user_id = @user.id
```
Comment (maxjost) [inline app/controllers/reservations_controller.rb]: @reservation.user = current_user

File: app/controllers/reservations_controller.rb (hunk 1 above)
Comment (Belaxyx) [reply app/controllers/reservations_controller.rb]: done

File: app/controllers/reservations_controller.rb (hunk 2 above)
Comment (Belaxyx) [reply app/controllers/reservations_controller.rb]: ok

File: app/controllers/reservations_controller.rb (hunk 3 above)
Comment (Belaxyx) [reply app/controllers/reservations_controller.rb]: done

File: app/controllers/reservations_controller.rb (hunk 4 above)
Comment (Belaxyx) [reply app/controllers/reservations_controller.rb]: done

File: app/controllers/reservations_controller.rb (hunk 6 above)
Comment (Belaxyx) [reply app/controllers/reservations_controller.rb]: done

File: app/controllers/reservations_controller.rb (hunk 5 above)
Comment (Belaxyx) [reply app/controllers/reservations_controller.rb]: done

**Pull request closed without merging**